actions/bulk_link_issue.py

```diff
@@ -1,21 +1,44 @@
+import threading
+from threading import Semaphore
 from lib.base import BaseJiraAction
 
-__all__ = [
-    'BulkLinkJiraIssueAction'
-]
-
 
 class BulkLinkJiraIssueAction(BaseJiraAction):
+    def link_issues(
+        self,
+        semaphore,
+        issue_key=None,
+        target_issue=None,
+        direction=None,
+        link_type=None,
+    ):
+        with semaphore:
+            if direction == "outward":
+                outward_issue_key = issue_key
+                inward_issue_key = target_issue
+                response = self._client.create_issue_link(
+                    link_type, inward_issue_key, outward_issue_key
+                )
+
+            if direction == "inward":
+                inward_issue_key = issue_key
+                outward_issue_key = target_issue
+                response = self._client.create_issue_link(
+                    link_type, inward_issue_key, outward_issue_key
+                )
+            response_output = {"issue": target_issue, "response": response}
```

Comment:
In the response output can we add both the inward and outward issue targets? In testing this I just see something like this on my end with an outward link which just tells me i successfully linked some issue to my target:

```
{
  "stdout": "{'issue': 'NETOPS-1', 'response': <Response [201]>}
{'issue': 'NETOPS-1', 'response': <Response [201]>}
{'issue': 'NETOPS-1', 'response': <Response [201]>}
",
  "stderr": "",
  "exit_code": 0,
  "result": null
}
```

If one of the links fails and gets a non-201 response it is very difficult to figure out which one.
So thinking we should change this line to something like:

```
response_output = {"inward_issue": inward_issue_key, "outward_issue": outward_issue_key, "response": response}
```

To support this we probably need to initialize above line 16 something like:

```
outward_issue_key = ""
inward_issue_key = ""
```

Just to cover the case where direction isn't one of "inward" or "outward"

Reply:
Thanks @floatingstatic . I have made the requested chanegs.

The output after test look like below now:

```
{'inward_issue': 'DEVOPS-387', 'outward_issue': 'DEVOPS-394', 'response': <Response [201]>}
{'inward_issue': 'DEVOPS-384', 'outward_issue': 'DEVOPS-394', 'response': <Response [201]>}
[expand](https://st2-karam.bitovi-sandbox.com/#/code/result/65329d7067c65fdc8d4f6986
```)

Let me know if there is any other change you wnat me to make.

Reply:
Excellent I see the same +1